config.m4

```diff
@@ -75,8 +75,12 @@ if test "$PHP_MONGODB" != "no"; then
     PHP_CHECK_GCC_ARG(-Wno-unused-but-set-variable,     _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wno-unused-but-set-variable")
     PHP_CHECK_GCC_ARG(-Wno-missing-field-initializers,  _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wno-missing-field-initializers")
 
+    AC_SUBST(MONGOC_ENABLE_DEBUG_ASSERTIONS, 1)
+
     MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS"
     STD_CFLAGS="-g -O0 -Wall"
+  else
+    AC_SUBST(MONGOC_ENABLE_DEBUG_ASSERTIONS, 0)
```

Comment:
This is only relevant when we're using bundled libmongoc. I'd propose moving this down further in the file, akin to what we do for:

https://github.com/mongodb/mongo-php-driver/blob/5cfc495a195b8fd6a3d2afd28052608fb0a0b0cb/config.m4#L351-L355

Reply:
Good point. Done!